### PR TITLE
Fix the highlighter in console and debugger

### PR DESCRIPTION
**Pull request by hbenl**

### CODE REVIEW

**Review by jasonLaster — APPROVED** [review]

File: src/devtools/client/webconsole/actions/toolbox.js [hunk 1]
```diff
@@ -10,10 +10,16 @@ import { paintGraphicsAtTime } from "protocol/graphics";
...
+  return async ({ toolbox }) => {
+    if (grip.getPause() !== ThreadFront.currentPause) {
+      return;
+    }
+
     const highlighter = toolbox.getHighlighter();
-    if (highlighter) {
-      highlighter.highlight(grip);
+    const nodeFront = grip.getNodeFront();
+    if (highlighter && nodeFront) {
```
Comment (jasonLaster) [inline src/devtools/client/webconsole/actions/toolbox.js:20]: when will highlighter not be available?

File: src/devtools/client/debugger/src/actions/toolbox.js [hunk 2]
```diff
@@ -33,8 +33,10 @@ export function openInspector(grip) {
 }
 
 export function highlightDomElement(grip) {
-  return ({ toolbox }) => {
-    toolbox.getPanel("debugger")?.highlightDomElement(grip);
+  return async ({ toolbox }) => {
+    const nodeFront = grip.getNodeFront();
+    await nodeFront.ensureLoaded();
+    toolbox.getPanel("debugger")?.highlightDomElement(nodeFront);
```
Comment (jasonLaster) [inline src/devtools/client/debugger/src/actions/toolbox.js:39]: nit, could we have one shared action?

**Review by jasonLaster — COMMENTED** [review]
This would be nice to have an e2e test.

**Comment by jasonLaster** [comment]
I think this should be safe to delete

```diff
diff --git a/src/devtools/client/debugger/panel.js b/src/devtools/client/debugger/panel.js
index 3e23e2b8..2630b349 100644
--- a/src/devtools/client/debugger/panel.js
+++ b/src/devtools/client/debugger/panel.js
@@ -61,15 +61,7 @@ export class DebuggerPanel {
     });
   }

-  async highlightDomElement(gripOrFront) {
-    if (!this._highlight) {
-      const { highlight, unhighlight } = this.toolbox.getHighlighter();
-      this._highlight = highlight;
-      this._unhighlight = unhighlight;
-    }
-
-    return this._highlight(gripOrFront);
-  }
+

   unHighlightDomElement() {
     if (!this._unhighlight) {
diff --git a/src/devtools/client/debugger/src/actions/toolbox.js b/src/devtools/client/debugger/src/actions/toolbox.js
index c287fe38..29e9ade4 100644
--- a/src/devtools/client/debugger/src/actions/toolbox.js
+++ b/src/devtools/client/debugger/src/actions/toolbox.js
@@ -32,12 +32,6 @@ export function openInspector(grip) {
   };
 }

-export function highlightDomElement(grip) {
-  return ({ toolbox }) => {
-    toolbox.getPanel("debugger")?.highlightDomElement(grip);
-  };
-}
-
 export function unHighlightDomElement(grip) {
```

**Comment by jasonLaster** [comment]
Also looks like `inspector-02` could be updated to also test highlighting from the console

File: src/devtools/client/webconsole/actions/toolbox.js (hunk 1 above)
Comment (hbenl) [reply src/devtools/client/webconsole/actions/toolbox.js:20]: I don't know, I just assumed the check was there for a reason. And it doesn't hurt, so I didn't want to take any chances.

File: src/devtools/client/debugger/src/actions/toolbox.js (hunk 2 above)
Comment (hbenl) [reply src/devtools/client/debugger/src/actions/toolbox.js:39]: I didn't want to start a refactoring.

**Comment by hbenl** [comment]
I'll write an e2e test in another PR.

**Pull request merged**